libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/MapboxCarContext.kt

```diff
@@ -1,37 +1,101 @@
 import com.mapbox.maps.extension.androidauto.MapboxCarMap
-import kotlinx.coroutines.flow.MutableStateFlow
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 
+/**
+ * This is the entry point for Mapbox Navigation in Android Auto. Create this object and then you
+ * have access to services and options for customization.
+ *
+ * @param lifecycle used to access the [CarContext].
+ * @param mapboxCarMap controls the Mapbox car map surface.
+ */
 @OptIn(MapboxExperimental::class)
 class MapboxCarContext(
-    val carContext: CarContext,
-    val lifecycleOwner: LifecycleOwner,
+    /**
+     * Gives access to the [Lifecycle].
+     */
+    val lifecycle: Lifecycle,
+    /**
+     * [MapboxCarMap] controls the Mapbox car map surface
+     */
     val mapboxCarMap: MapboxCarMap,
-    val feedbackPollProvider: CarFeedbackPollProvider = CarFeedbackPollProvider(),
-    val carFeedbackOptions: CarFeedbackOptions = CarFeedbackOptions.Builder().build(),
-    routeOptionsInterceptor: CarRouteOptionsInterceptor = CarRouteOptionsInterceptor { it },
-    val carPlaceSearchOptions: CarPlaceSearchOptions = CarPlaceSearchOptions.Builder().build(),
 ) {
-    val carSettingsStorage = CarSettingsStorage(carContext)
-    val mapboxScreenManager = MapboxScreenManager(carContext, lifecycleOwner)
+    private val carContextOwner = MapboxCarContextOwner(lifecycle)
 
-    val speedLimitOptions = MutableStateFlow(SpeedLimitOptions.Builder().build())
+    /**
+     * Mutable options
+     */
+    val options: MapboxCarOptions = MapboxCarOptions()
 
-    val carRoutePreviewRequest = CarRoutePreviewRequest(routeOptionsInterceptor)
+    /**
+     * Gives access to the [CarContext].
+     */
+    val carContext: CarContext by mapboxCarService("CarContext") {
+        carContextOwner.carContext()
+    }
+
+    /**
+     * Control the screens shown with the [MapboxScreenManager]
+     */
+    val mapboxScreenManager = MapboxScreenManager(carContextOwner)
+
+    /**
+     * Inorder to enable this service must be registered to the [MapboxNavigationApp]
+     *
+     * @see MapboxCarNavigationManager
+     * @see NavigationManager
+     */
+    val mapboxNavigationManager by mapboxCarService("MapboxNavigationManager") {
+        MapboxCarNavigationManager(carContext)
+    }
+
+    /**
+     * Inorder to enable this service must be registered to the [MapboxNavigationApp]
+     *
+     * @see MapboxCarNavigationManager
+     * @see NavigationManager
+     */
+    val notificationInterceptor: CarNotificationInterceptor by mapboxCarService(
+        "MapboxNotificationInterceptor"
+    ) {
+        val carAppServiceClass = options.notificationTargetService
+        checkNotNull(carAppServiceClass) {
+            "MapboxNotificationInterceptor cannot be used unless the " +
+                "options.notificationCarSppService is set to your CarAppService"
+        }
+        CarNotificationInterceptor(carContext, carAppServiceClass)
+    }
```

Comment:
What is the reason for having this in `MapboxCarContext`? Developers still have to register it manually. I think it is more convenient to instantiate `CarNotificationInterceptor` manually, rather that set `notificationTargetService` via `MapboxCarOptions` and then get `notificationInterceptor` from `MapboxCarContext`. 
The same applies to `mapboxNavigationManager`.

Reply:
I'm thinking of MapboxCarContext as an entry point to the SDK. Similar to how you can access `CarContext.getCarServiceName`. This way a developer can look here and discover features or services.